Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -174,9 +182,69 @@ namespace OpenInfraPlatform
 
 					BLUE_LOG(warning) << "Geometry conversion for IfcSectionedSolidHorizontal not implemented.";
 					// TO DO: implement, check for formal propositions. 
+                 
 
-				 //endif sectioned_solid_horizontal
-			}
+				//check dimensions and correct attributes sizes
+                if(vec_cross_sections.size() != vec_cross_section_positions.size())
+				{
+					std::cout << "CrossSections and CrossSectionsPositions are not equal in size " << std::endl;
+					return;
+				}
+				if ((vec_cross_sections.size() - vec_cross_section_positions.size()) > 1)
+				{
+					std::cout << "vec_cross_sections size decreased by one" << std::endl;
+					vec_cross_sections.pop_back();
+				}
+				else if ((vec_cross_section_positions.size() - vec_cross_sections.size()) > 1)
+				{
+					std::cout << "vec_cross_section_positions decreased by one" << std::endl;
+					vec_cross_section_positions.pop_back();
+				}
+
+                 //TO DO: check lenght conversions on the ProfileDistanceExpression L[2:?] CrossSectionPostitions
+				double length_in_meter = UnitConvert()->getLengthInMeterFactor();
+
+
+				 //Give directrix to Curve converter
+		    	 std::vector<carve::geom::vector<3> > segment_start_points;
+	     		 std::vector<carve::geom::vector<3> > basis_curve_points;
+				 curveConverter->convertIfcCurve(directrix.lock(), basis_curve_points, segment_start_points);
+
+				 std::shared_ptr<carve::input::PolyhedronData> pipe_data(new carve::input::PolyhedronData());
+				 itemData->closed_polyhedrons.push_back(pipe_data);
+				 //std::vector<carve::geom::vector<3> > inner_shape_points;  //TO DO: find out if i need inner_shape_points for the CrossSections
+
+				 int num_curve_points = basis_curve_points.size();
+				 carve::math::Matrix matrix_sweep;
+
+				 // Less than two points is a point
+				 if (num_curve_points < 2)
+				 {
+					 std::cout << "IfcSectionedSolidHoritzontal: num curve points < 2" << std::endl;
+					 return;

```

Comment:
Again, a `throw` would be more suitable.

Reply:
resolved in commit ccd8948